src/main/kotlin/no/nav/klage/oppgave/api/KlagebehandlingFacade.kt

```diff
@@ -21,11 +30,40 @@ class KlagebehandlingFacade(
         private val securelogger = getSecureLogger()
     }
 
-    fun getKlagebehandling(oppgaveId: Long): KlagebehandlingView {
+    fun getKlagebehandling(klagebehandlingId: UUID): KlagebehandlingView {
         return klagebehandlingMapper.mapKlagebehandlingToKlagebehandlingView(
-            klagebehandlingService.getKlagebehandlingByOppgaveId(
-                oppgaveId
+            klagebehandlingService.getKlagebehandling(klagebehandlingId)
+        )
+    }
+
+    fun searchOppgaver(oppgaverSearchCriteria: OppgaverSearchCriteria): OppgaverRespons {
+        val esResponse = elasticsearchRepository.findByCriteria(oppgaverSearchCriteria)
+        return OppgaverRespons(
+            antallTreffTotalt = esResponse.totalHits.toInt(),
+            oppgaver = oppgaveMapper.mapEsKlagebehandlingerToView(
+                esResponse.searchHits.map { it.content },
+                oppgaverSearchCriteria.isProjectionUtvidet()
             )
         )
     }
+
+    fun assignOppgave(klagebehandlingId: UUID, saksbehandlerIdent: String?, klagebehandlingVersjon: Long) {
+        klagebehandlingService.assignOppgave(klagebehandlingId, klagebehandlingVersjon, saksbehandlerIdent)
+        val oppgaveIderForKlagebehandling = klagebehandlingService.getOppgaveIderForKlagebehandling(klagebehandlingId)
```

Comment:
Kan flere oppgaver tilhøre samme behandling? Akkurat som en oppgave kan ha mange behandlinger? Har vi faktisk mange til mange her?

Reply:
Modellen er jo mange til mange, men konseptuelt er det vel en til mange. Jeg tror jeg har lagt inn en kommentar på det, at der vi har en liste så er det egentlig bare en oppgave i lista. Jeg vurderte å skrive det om, men tenkte det ble for mye. Vi har jo en oneToMany relasjon i JPA nå, men ønsker egentlig en oneToOne. Men i databasen er fremmednøklene lagt slik at det passer best med oneToMany. Jeg googla litt, og det vanlige i JPA-land er å beholde en oneToMany i de tilfellene, og så heller "skjule det" ved at gettere og settere behandler det som 1 og lagrer det som en singletonList. Det kan vi jo sikkert gjøre i Kotlin også, men jeg har ikke gjort det ennå.